src/benchmark/managers/config.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""The config manager class.
+
+This class summarizes all the configuration needed for the workload execution
+and returns a model containing that information.
+"""
+
+import logging
+import os
+import time
+from abc import abstractmethod
+from typing import Any, Optional
+
+from jinja2 import DictLoader, Environment, FileSystemLoader, exceptions
+
+from benchmark.core.models import (
+    DPBenchmarkWrapperOptionsModel,
+)
+from benchmark.core.workload_base import WorkloadBase
+from benchmark.events.db import DatabaseRelationHandler
+from benchmark.events.peer import PeerRelationHandler
```

Comment:
**issue** A manager cannot depend on events. The chain of dependency is core > managers > handlers